### PR TITLE
Working skeleton of navigation for reconsent flow

### PR DESCRIPTION
**Pull request by juliatan**
# Description

DO NOT MERGE -> DashboardScreen auto redirects to reconsent flow atm. 

https://www.notion.so/joinzoe/Product-Eng-Board-ca9e8b1b8926481f8c94cb01a88d567d?p=bd86d3de1b0a44ddba4410818c253734

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if needed)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

_Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/features/reconsent/components/ReconsentFooter.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+interface IProps {
+  onPress: () => void;
+  title: string;
+}
+
+function ReconsentFooter(props: IProps) {
+  const theme = useTheme();
+  return (
+    <View
+      style={{
```
Comment (aikewoody) [inline src/features/reconsent/components/ReconsentFooter.tsx:16]: This view could be removed by moving the style to the Button component and adding the padding to the margin

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { SafeLayout, Text } from '@covid/components';
+import i18n from '@covid/locale/i18n';
+import NavigatorService from '@covid/NavigatorService';
+import { styling, useTheme } from '@covid/themes';
+import { colors } from '@theme';
+import * as React from 'react';
+import { ScrollView, StyleSheet, View } from 'react-native';
+
+import ReconsentFooter from '../components/ReconsentFooter';
+import ReconsentHeader from '../components/ReconsentHeader';
```
Comment (aikewoody) [inline src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:10]: Let's use absolute import paths :)

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+export default function ReconsentDiseaseConfirmationScreen() {
+  const theme = useTheme();
+
+  return (
+    <SafeLayout style={styles.page}>
+      <ScrollView contentContainerStyle={styling.flexGrow} style={{ paddingHorizontal: theme.grid.gutter }}>
+        <ReconsentHeader showBackIcon showDots />
+        <View>
```
Comment (aikewoody) [inline src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:19]: This view looks obsolete

**Comment by aikewoody** [comment]
Looks great 👍  Let's implement those screens!

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx (hunk 2 above)
Comment (bionara) [reply src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:10]: Does our linter pick these up btw?

File: src/features/reconsent/components/ReconsentFooter.tsx (hunk 1 above)
Comment (bionara) [reply src/features/reconsent/components/ReconsentFooter.tsx:16]: Maybe worth us making a linting rule to detect <View> and/or <View style..

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx (hunk 2 above)
Comment (aikewoody) [reply src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:10]: Not yet but it should!

File: src/features/reconsent/components/ReconsentFooter.tsx (hunk 1 above)
Comment (juliatan) [reply src/features/reconsent/components/ReconsentFooter.tsx:16]: I don't love the idea of adding margin and padding to components like `Button`. Makes the `Button` less reusable?
Unless you just mean adding it to line 27?

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx (hunk 2 above)
Comment (juliatan) [reply src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:10]: Just to check what you mean by this - is it the creation of an `index.ts` file for each folder? And is the benefit just that it's shorter paths when importing?

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx (hunk 3 above)
Comment (juliatan) [reply src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:19]: It's placeholder at the moment to wrap all of a screen's content... so up to whoever is working on each screen to edit. :)

**Comment by juliatan** [comment]
> Looks great 👍 Let's implement those screens!

Cool. QQ: Do you think we need to create (yet another!) layout component e.g. `ReconsentScreen` to keep all the boilerplate layout components like `SafeLayout` and `ScrollView`?

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx (hunk 3 above)
Comment (bionara) [reply src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:19]: 🧹

**Review by bionara — DISMISSED** [review]
Looks great so far!

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx (hunk 2 above)
Comment (aikewoody) [reply src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:10]: Converting this to absolute imports would be:
`import ReconsentFooter  from '@covid/features/reconsent/components/ReconsentFooter';`
`import ReconsentFooter  from '@covid/features/reconsent/components/ReconsentHeader';`

File: src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx (hunk 3 above)
Comment (aikewoody) [reply src/features/reconsent/screens/ReconsentDiseaseConfirmationScreen.tsx:19]: Got it 👍

**Comment by aikewoody** [comment]
> > Looks great 👍 Let's implement those screens!
> 
> Cool. QQ: Do you think we need to create (yet another!) layout component e.g. `ReconsentScreen` to keep all the boilerplate layout components like `SafeLayout` and `ScrollView`?

100% yes! I'll do that when creating the introduction screen

**Pull request closed without merging**